File: java/leetcode/max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int consec = 0;
        int maxConsec = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 1) {
                consec++;
            } else {
                if (consec > maxConsec) { maxConsec = consec; }
                consec = 0;
            }
        }
        if (consec > maxConsec) { maxConsec = consec; }
        return maxConsec;
    }
}

File: python/leetcode/max-consecutive-ones.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        consec = 0
        max_consec = 0
        for i in range(len(nums)):
            if nums[i] == 1:
                consec += 1
            else:
                if consec > max_consec:
                    max_consec = consec
                consec = 0
        if consec > max_consec:
            max_consec = consec
        return max_consec
